src/game.cpp

```diff
@@ -8015,47 +8016,58 @@ bool game::vehicle_near()
     return false;
 }
 
-bool game::refill_vehicle_part(vehicle &veh, vehicle_part *part, bool test)
+bool game::refill_vehicle_part(vehicle &veh, vehicle_part *part_ptr, bool test)
 {
-    vpart_info part_info = vehicle_part_types[part->id];
-    if (!part_info.has_flag("FUEL_TANK")) {
+(void)veh;(void)part_ptr;(void)test;
+    return false;// TODO: remove function
+}
+
+bool game::pl_refill_vehicle(vehicle &veh, int part, bool test)
+{
+    (void)test; // TODO: remove unused variable
```

Comment:
FYI: this is actually a very nice way to keep the logic in one function:
```C++
if( !pl_refill_vehicle( ..., true ) ) {
    popup("You can't do that!");
    // or just deactivate the option in the menu
    return;
}
pl_refill_vehicle( ..., false ); // With the very same ... as above.
```
This keeps the logic to check whether something can be done and the logic to do the thing neatly together. You can however make this function private and add two wrapper functions (aptly named): one that does the actual refilling (forwards to `refill(..., false)`) and one that tests only (forwards to `refill(..., true)`).

Any changes to the logic will be consistent to the check and the actual processing function.